src/odbc/connectparams.c

```diff
@@ -268,6 +286,205 @@ odbc_dstr_swap(DSTR *a, DSTR *b)
 	*b = tmp;
 }
 
+static bool
+odbc_parse_connect_string_value_escaped(TDS_ERRS *errs, const char **current_position, const char **current_end, const char *connect_string_end, DSTR *value)
+{
+	const char *end;
+	char *new_value = strdup(*current_position);
```

Comment:
`strdup` can fail. Why do you need to call it by the way? Won't it be easier to duplicate all at the beginning and make the string writable?